CONTRIBUTING.md

```diff
@@ -1,183 +1,71 @@
+This guide is intended for members of the ISPyB collaboration with contributor access to the repository. If you are not a member but would like to contribute, please contact us.
 
-An example how to describe a class:
+-   [Reporting Bugs](#reporting-bugs)
+-   [Submiting code to the repository](#submiting-code-to-the-repository)
+-   [Reviewing process](#reviewing-process)
+-   [Coding style guidelines](#coding-style-guidelines)
 
-```bash
-class ExampleClass(object):
-  """The summary line for a class docstring should fit on one line.
-
-  If the class has public attributes, they may be documented here
-  in an ``Attributes`` section and follow the same formatting as a
-  function's ``Args`` section. Alternatively, attributes may be documented
-  inline with the attribute's declaration (see __init__ method below).
-
-  Properties created with the ``@property`` decorator should be documented
-  in the property's getter method.
-
-  Attributes:
-      attr1 (str): Description of `attr1`.
-      attr2 (:obj:`int`, optional): Description of `attr2`.
-
-  """
-
-  def __init__(self, param1, param2, param3):
-      """Example of docstring on the __init__ method.
+### Reporting bugs
 
-      The __init__ method may be documented in either the class level
-      docstring, or as a docstring on the __init__ method itself.
+Before submitting a new bug check if the bug is not already reported in the [bug issues](https://github.com/ispyb/py-ispyb/issues?q=is%3Aopen+is%3Aissue+label%3Abug).
+If the corresponding issue does not exist then:
 
-      Either form is acceptable, but the two should not be mixed. Choose one
-      convention to document the __init__ method and be consistent with it.
+-   [Open a new issue](https://github.com/ispyb/py-ispyb/issues/new) with a short description in the title.
+-   In the description describe the bug:
+    -   Conditions when the bug appears;
+    -   How it can be reproduced;
+    -   Possible cause of the bug and source code where it occurs;
+    -   If possible add an error log.
+-   Assign the `bug` label to the issue.
 
-      Note:
-          Do not include the `self` parameter in the ``Args`` section.
+### Submitting code to the repository
 
-      Args:
-          param1 (str): Description of `param1`.
-          param2 (:obj:`int`, optional): Description of `param2`. Multiple
-              lines are supported.
-          param3 (list(str)): Description of `param3`.
+To submit code to the repository, please follow these steps:
 
-      """
-      self.attr1 = param1
-      self.attr2 = param2
-      self.attr3 = param3  #: Doc comment *inline* with attribute
+1. All code contribution has to be done from an issue. If there is no existing issue for the submission you wish to make, start by [creating a new one](https://github.com/ispyb/py-ispyb/issues/new), and describe what should be done for it to be considered as fulfilled;
```

Comment:
Sounds a bit heavyweight for a project with so few developers. But if people are happy with it, sure. Just make sure that you actually enforce it, nothing worse than rules that are ignored.

Reply:
The idea behind that is that we avoid having unaffiliated branches in the repository for which only the author knows what's going on. If we always create an issue, and always create a branch from the issue, then all branches are linked to an issue, and we know what they are here for.
Of course this can't happen if we say you can contribute without an issue.